### PR TITLE
Change AWS::Batch::ComputeResources.Tags type to dict

### PR DESCRIPTION
**Pull request by flou**
AWS Batch seems to use a different Tag format. Instead of the usual
```
{
  'Key': 'TagKey',
  'Value': 'TagValue'
}
```
AWS Batch expects a regular map `{'TagKey': 'TagValue'}`.

This commit also adds an example template for AWS Batch.

### CODE REVIEW

File: troposphere/batch.py [hunk 1]
```diff
@@ -10,13 +10,13 @@ class ComputeResources(AWSProperty):
         "SecurityGroupIds": ([basestring], True),
         "BidPercentage": (positive_integer, False),
         "Type": (basestring, True),
-        "Subnets": ([basestring], True),
+        "Subnets": (list, True),
```
Comment (markpeek) [inline troposphere/batch.py]: Any reason why this was changed? Using [basestring] is a list of basestrings which is more specific than just using list.

File: troposphere/batch.py [hunk 2]
```diff
@@ -10,13 +10,13 @@ class ComputeResources(AWSProperty):
...
         "MinvCpus": (positive_integer, True),
         "ImageId": (basestring, False),
         "InstanceRole": (basestring, True),
         "InstanceTypes": ([basestring], True),
         "Ec2KeyPair": (basestring, False),
-        "Tags": ([basestring], False),
+        "Tags": (dict, False),
```
Comment (markpeek) [inline troposphere/batch.py]: The [documentation](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-batch-computeenvironment-computeresources.html#cfn-batch-computeenvironment-computeresources-tags) is all over the map on this one as it is referring to the Tags as being normal [Tags](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html). I'll have to follow up with my AWS contacts to understand why there is a discrepancy.

File: troposphere/batch.py (hunk 1 above)
Comment (flou) [reply troposphere/batch.py]: In a previous version of this PR, the example template used a `CommaDelimitedList` for the `Subnets` parameter type and the build failed for Python 2.7 (see https://travis-ci.org/cloudtools/troposphere/jobs/292160894). Not sure why it failed though…

So I figured I'd use the `list` property type as in `AWS::RDS::DBSubnetGroup.SubnetIds` or `AWS::ElasticLoadBalancing::LoadBalancer.Subnets`.

File: troposphere/batch.py (hunk 2 above)
Comment (flou) [reply troposphere/batch.py]: I understand. It took me a while to figure out how AWS Batch tags work, but when using normal Tags CloudFormation gives the following message:
```
Property validation failure: [Value of property {/ComputeResources/Tags} does not match type {Map}]
```
which is very weird when comparing with the docs.

Comment (flou) [reply troposphere/batch.py]: Do you have any news about this?

**Comment by markpeek** [comment]
Thanks! (Note: I did revert the Subnet change)

**Pull request merged**